libsql_client/hrana/conn.py

```diff
@@ -304,6 +339,37 @@ def get_result(response: proto.Response) -> proto.BatchResult:
             return cast(proto.BatchResp, response)["result"]
         return _map_future(response_fut, get_result)
 
+    def store_sql(self, sql: str) -> asyncio.Future[int]:
+        if self._state.closed is not None:
+            raise LibsqlError("Stream was closed", "STREAM_CLOSED") from self._state.closed
+
+        sql_id = self._sql_id_alloc.alloc()
+        request: proto.StoreSqlReq = {
+            "type": "store_sql",
+            "sql_id": sql_id,
+            "sql": sql,
+        }
+        response_fut = self._conn.send_request(request)
+
+        def get_result(response: proto.Response) -> int:
+            return sql_id
+        return _map_future(response_fut, get_result)
+
+    def close_sql(self, sql_id: int) -> asyncio.Future[None]:
+        if self._state.closed is not None:
+            raise LibsqlError("Stream was closed", "STREAM_CLOSED") from self._state.closed
+
+        request: proto.CloseSqlReq = {
+            "type": "close_sql",
+            "sql_id": sql_id,
+        }
+        response_fut = self._conn.send_request(request)
+
+        def get_result(response: proto.Response) -> None:
+            self._sql_id_alloc.free(sql_id)
+            return None
+        return _map_future(response_fut, get_result)
```

Comment:
This method also does not need to be async.

Reply:
good catch, done.